java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -100,4 +96,82 @@ protected static ConnectionManager buildConnectionManager(ChannelHandler channel
     protected static CommandManager buildCommandManager(ChannelHandler channelHandler) {
         return new CommandManager(channelHandler);
     }
+
+    /**
+     * Extracts the response from the Protobuf response and either throws an exception or returns the
+     * appropriate response as an <code>Object</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response <code>Object</code>
+     */
+    protected Object handleObjectResponse(Response response) {
+        // convert protobuf response into Object
+        return new BaseCommandResponseResolver(RedisValueResolver::valueFromPointer).apply(response);
+    }
+
+    /**
+     * Extracts the response value from the Redis response and either throws an exception or returns
+     * the value as a <code>String</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response as a <code>String</code>
+     * @throws RedisException if there's a type mismatch
+     */
+    protected String handleStringResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value instanceof String || value == null) {
+            return (String) value;
+        }
+        throw new RedisException(
+                "Unexpected return type from Redis: got "
+                        + value.getClass().getSimpleName()
+                        + " expected String");
+    }
+
+    /**
+     * Extracts the response value from the Redis response and either throws an exception or returns
+     * the value as an <code>Object[]</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response as an <code>Object[]</code>
+     * @throws RedisException if there's a type mismatch
+     */
+    protected Object[] handleArrayResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value instanceof Object[]) {
+            return (Object[]) value;
+        }
+        String className = (value == null) ? "null" : value.getClass().getSimpleName();
+        throw new RedisException(
+                "Unexpected return type from Redis: got " + className + " expected Object[]");
+    }
+
+    /**
+     * Extracts the response value from the Redis response and either throws an exception or returns
+     * the value as a <code>Map</code>.
+     *
+     * @param response Redis protobuf message
+     * @return Response as a <code>Map</code>
+     * @throws RedisException if there's a type mismatch
+     */
+    @SuppressWarnings("unchecked")
+    protected Map<Object, Object> handleMapResponse(Response response) {
+        Object value = handleObjectResponse(response);
+        if (value instanceof Map) {
+            return (Map<Object, Object>) value;
+        }
+        String className = (value == null) ? "null" : value.getClass().getSimpleName();
+        throw new RedisException(
+                "Unexpected return type from Redis: got " + className + " expected Map");
+    }
+
```

Comment:
Maybe we should make sections/regions - for handlers, for commands for all rest

Reply:
Sure.  Especially as we add a ton of commands.